sourced/ml/transformers/basic.py

```diff
@@ -277,22 +275,24 @@ def create_parquet_loader(session_name, repositories,
     return parquet
 
 
-def create_uast_source(args, session_name, select=HeadFiles, language_selector=None,
-                       extract_uast=True):
+def create_uast_source(args, session_name, select=HeadFiles, extract_uast=True):
```

Comment:
I find it weird that the function is named `create_uast_source` and it actually has a possibility to not extract UASTs because of `extract_uast=False`. Let's extract the common part which does not depend on `extract_uast` to new function `create_file_source` and use it from `create_uast_source`.